src/libstore/builtins/buildenv.cc

```diff
@@ -13,6 +13,30 @@ struct State
     unsigned long symlinks = 0;
 };
 
+/**
+ * This exception is private and only used within buildenv.cc.
+ * Its purpose is to pass-through contextual error information.
+ * It inherits from std::exception instead of nix::Error to
+ * avoid having to render pretty errors.
+ */
+class BuildEnvFileConflictError : public std::exception
+{
+public:
+    const Path originalFile;
+    const Path conflictingFile;
+    int priority;
+
+    BuildEnvFileConflictError(
+        const Path originalFile,
+        const Path conflictingFile,
+        int priority
+    )
+        : originalFile(originalFile)
+        , conflictingFile(conflictingFile)
+        , priority(priority)
+    {}
```

Comment:
It would be nicer to use designated initializers, e.g.
```
BuildEnvFileConflictError {
    .originalFile = ...,
    .conflictingFile = ...,
    .priority = ...
}
```
which is more readable and removes the need for a constructor.

Reply:
Hmm, I looked into this, is this even possible when it is inheriting from std::exception?

Not entirely sure, but it seems it is mostly possible for structs?

```
src/libstore/builtins/buildenv.cc:113:25: error: designated initializers cannot be used with a non-aggregate type ‘nix::BuildEnvFileConflictError’
```

Reply:
Ah maybe `std::exception` isn't really needed, and then this can be a simple plain old struct.

Reply:
Aah, yes, without using `std::exception` I could indeed change it to struct, which works for throwing :+1: